### PR TITLE
[SYCL] Generalise CPU selector to non-Intel CPUs.

### PR DESCRIPTION
**Pull request by j-stephan**
I cherry picked a large commit from #1845 in order to support a later PR which factors out RNG from #1845.

**Please merge #1981 first.**

### CODE REVIEW

**Comment by fwyzard** [comment]
Rebased on top of `develop` to make GitHub happy - [no changes](https://github.com/alpaka-group/alpaka/compare/92ef169be38b8fdc1b0e76529e2fbdfefea1619a..0687bc7b262490bb6274f22be482fc108c41dcad).

File: include/alpaka/test/Check.hpp [hunk 1]
```diff
@@ -14,7 +14,16 @@
...
+                                 decltype(acc),                                                                       \
+                                 alpaka::TagCpuSycl,                                                                  \
+                                 alpaka::TagFpgaSyclIntel,                                                            \
+                                 alpaka::TagFpgaSyclXilinx,                                                           \
+                                 alpaka::TagGpuSyclIntel,                                                             \
+                                 alpaka::TagGenericSycl>)                                                             \
+                    acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                           \
+                else                                                                                                  \
+                    printf("ALPAKA_CHECK failed because '!(%s)'\n", #expression);*/                                   \
```
Comment (fwyzard) [inline include/alpaka/test/Check.hpp:26]: Should we just use `printf` unconditionally ?

**Comment by fwyzard** [comment]
Please merge #1985 before (rebasing and) merging this one.

**Comment by fwyzard** [comment]
It turns out that rebasing #1845 on top of this PR is significantly more complicated.
I would propose to delay or close this PR until the work in #1845 is completed, and can be factored out more easily.

**Comment by bernhardmgruber** [comment]
> I would propose to delay or close this PR until the work in https://github.com/alpaka-group/alpaka/pull/1845 is completed, and can be factored out more easily.

I have put the PR in draft mode, so the green check does not distract me. Please reactivate or close the PR as you please.

**Comment by j-stephan** [comment]
@fwyzard Also this one?

**Comment by fwyzard** [comment]
Yes.

**Pull request closed without merging**